solr/core/src/java/org/apache/solr/servlet/HttpSolrCall.java

```diff
@@ -1066,13 +1066,14 @@ private Map<String, Integer> checkStateVersionsAreValid(String stateVer) {
 
   protected SolrCore getCoreByCollection(String collectionName, boolean isPreferLeader) {
     ZkStateReader zkStateReader = cores.getZkController().getZkStateReader();
-
     ClusterState clusterState = zkStateReader.getClusterState();
-    DocCollection collection = clusterState.getCollectionOrNull(collectionName, true);
+    DocCollection collection = resolveDocCollection(collectionName);
+    // the usage of getCoreByCollection assumes that if null is returned, collection is found, but
+    // replicas might not
+    // have been created. Hence returning null here would be misleading...
```

Comment:
Seems like you could write javadocs for getCoreByCollection additionally or instead

Reply:
Done